picky-signtool/src/verify.rs

```diff
@@ -154,15 +158,16 @@ pub fn authenticode_signature_ps_from_file(file_path: &Path) -> anyhow::Result<A
     let mut buffer = Vec::new();
     file.read_to_end(&mut buffer).unwrap();
 
-    let (decoded, _) = encoding::decode(
-        &buffer,
-        DecoderTrap::Strict,
-        encoding::all::UTF_8 as encoding::EncodingRef,
-    );
-
-    let buffer = decoded.map_err(|err| anyhow!("Failed to decoded {:?} ps file: {}", file_path, err))?;
+    // invalid characters will be replaced with � (replacement character)
+    let (buffer, ..) = if buffer.starts_with(&UTF16_LE_BOM) {
+        encoding_rs::UTF_16LE.decode(&buffer)
+    } else if buffer.starts_with(&UTF16_BE_BOM) {
+        encoding_rs::UTF_16BE.decode(&buffer)
+    } else {
+        encoding_rs::UTF_8.decode(&buffer)
+    };
 
-    let signature = extract_ps_authenticode_signature(buffer)
+    let signature = extract_ps_authenticode_signature(buffer.to_string())
```

Comment:
If `extract_ps_authenticode_signature` absolutely requires an owned String, then use `into_owned` so that there is no copy if the value is already owned. But looking at the `extract_ps_authenticode_signature` function, I see this could take a `&str` instead.